deployments/zeppelin/bin/create-user-tools.sh

```diff
@@ -21,214 +21,200 @@
 
     createlinuxuser()
         {
-        local user="${1:?}"
-        local uid="${2}"
-        local gid="${3}"
-        local home="${4:-/home/${user}}"
-
-        #
-        # Call to Zeppelin node to create the Linux user account.
-        # The test for zero and the call to 'getnextuid' would be done on the Zeppelin node.
+        local username="${1:?'username required'}"
+        local usertype="${2:?'usertype required'}"
+        local uid="${3}"
+        local home="${4}"
         #
+        # Call Zeppelin to create the Linux user account.
+        # Returns JSON.
+        ssh zeppelin \
+            "
+            create_unix_user.sh '${username}' '${usertype}' '${uid}' '${home}'
+            "
+        }
 
-        if [ -z ${uid} ]
-        then
-            uid=$(getnextuid)
-        fi
-        if [ -z ${gid} ]
-        then
-            gid=${uid}
-        fi
-
-
-cat << EOF
-{
-"name": "${user}",
-"uid":  ${uid},
-"gid":  ${gid},
-"home": "${home}"
-}
-EOF
+    createhdfsspace()
+        {
+        local username="${1:?'username required'}"
+        local usertype="${2:?'usertype required'}"
+        local hdfsroot="${3:-'hdfs-${usertype}'}"
+        #
+        # Call Zeppelin to create the user's HDFS space.
+        # Returns JSON.
+        ssh zeppelin \
+            "
+            create_hdfs_user.sh '${username}' '${usertype}' '${hdfsroot}'
+            "
         }
 
-    createusershare()
+    createcephshare()
         {
-        local username="${1:?}"
-        local uid="${2:?}"
-        local gid="${3:?}"
-        local sharepath="${4:-/user/${user}}"
-        local sharesize="${5:-${defaultsharesize}}"
-        local sharename="user-data-${username}"
+        local cloudname="${1:?'cloudname required'}"
+        local username="${2:?'username required'}"
+        local usertype="${3:?'usertype required'}"
+        local uid="${4:?'user id required'}"
+
+        local cephroot="/ceph-${usertype}"
+        local sharepath="${5:-${cephroot}/${username}}"
+        local sharesize="${6:-${defaultsharesize}}"
+        local sharename="${usertype}-${username}"
         local shareuuid=$(uuidgen)
 
         #
         # Call to Openstack to create the share.
         #
+        create-ceph-share.sh \
+            "${cloudname}" \
+            "${sharename}" \
+            "${sharesize}"
+
 
         #
-        # Call to Zeppelin node to mount the share.
+        # Call to Zeppelin to mount the share.
         #
 
 cat << EOF
 {
 "name": "${sharename}",
 "uuid": "${shareuuid}",
 "path": "${sharepath}",
-"size": ${sharesize}
+"size":  ${sharesize}
 }
 EOF
         }
 
+    cloneusernotebooks()
+        {
+        local username="${1:?'username required'}"
+        local usertype="${2:?'usertype required'}"
+        local userpass="${3}"
+        #
+        # Call Zeppelin to clone the user's notebooks.
+        # Returns null (could return JSON list).
+        if [ -n "${userpass}" ]
+        then
+            ssh zeppelin \
```

Comment:
All of these tools should be run from the `ansible-client` container.

I agree, the location of these scripts isn't ideal. Moving `deployments/zeppelin/bin` to `deployments/ansibler/bin` would make this clearer. Perhaps in the next PR we should move things around so that it is clearer from the path where they should be executed.

Reply:
The `05-config-ssh.yml` playbook sets up a ssh config for `root` in the `ansible-client` container.

From inside the  `ansible-client` container, `cat ~/.ssh/config` should display something like this:
```
....
# Primary gateway via our Zeppelin node.
Host zeppelin
    User fedora
    HostName 128.232.222.76
    ControlPath ~/.ssh/%r@%h:%p
    ControlMaster auto
    ControlPersist 5m

# Master nodes.
Host master01
    User fedora
    ProxyCommand ssh -W %h:%p -l fedora -F /root/.ssh/ansible-config zeppelin
    ControlPath ~/.ssh/%r@%h:%p
    ControlMaster auto
    ControlPersist 5m

# Worker nodes.
Host worker01
    User fedora
    ProxyCommand ssh -W %h:%p -l fedora -F /root/.ssh/ansible-config zeppelin
    ControlPath ~/.ssh/%r@%h:%p
    ControlMaster auto
    ControlPersist 5m
....
```

You should be able to type things like this in the `ansible-client` container and they should _'just work'_.
```
ssh zeppelin
```
or
```
ssh worker01
```